core/trino-main/src/main/java/io/trino/execution/DenyTask.java

```diff
@@ -97,8 +96,7 @@ private static void executeDenyOnTable(Session session, Deny statement, Metadata
     {
         QualifiedObjectName tableName = createQualifiedObjectName(session, statement, statement.getName());
         RedirectionAwareTableHandle redirection = metadata.getRedirectionAwareTableHandle(session, tableName);
-        Optional<TableHandle> tableHandle = redirection.getTableHandle();
-        if (tableHandle.isEmpty()) {
+        if (redirection.getTableHandle().isEmpty()) {
```

Comment:
I feel this change unrelates to `Optional.orElseThrow` leverage. Is separated a commit over kill?

Reply:
it's related. 

The pattern
```
Optional<...> x = expr;
if (x.isEmpty()) {
  throw;
}
```
should generally be avoided and you should use `orElseThrow`.
You should **not**, however, use `orElseThrow` where the variable would become unused. I.e. you shouldn't use it, if your goal is to check existence of some optional value. In such cases, i inlined the expression